torchtitan/parallelisms/pipelining_utils.py

```diff
@@ -9,10 +9,61 @@
     ScheduleFlexibleInterleaved1F1B,
     ScheduleInterleaved1F1B,
 )
-from torch.distributed.pipelining.schedules import get_schedule_class
+
+from torch.distributed.pipelining.schedules import (
+    get_schedule_class,
+    PipelineScheduleMulti,
+    PipelineScheduleSingle,
+)
 from torchtitan.logging import logger
 
 
+def generate_split_points(job_config, pp_dim, model_config):
+    schedule_class = get_schedule_class(
+        job_config.experimental.pipeline_parallel_schedule
+    )
+    if issubclass(schedule_class, PipelineScheduleSingle):
+        num_stages_per_rank = 1
+    elif issubclass(schedule_class, PipelineScheduleMulti):
+        # Multi-stage schedules support more than 2 stages per rank, but this is the default if
+        # no pipeline split is specified
+        num_stages_per_rank = 2
+    else:
+        raise ValueError(
+            f"Unsupported pipeline schedule: {job_config.experimental.pipeline_parallel_schedule}"
+        )
+    total_stages = pp_dim * num_stages_per_rank
+    num_layers = model_config.n_layers
+    if total_stages > num_layers:
+        raise ValueError("Total stages cannot be greater than the number of layers")
+
+    base_interval = num_layers // total_stages
+    remainder = num_layers % total_stages
+
+    # Calculate the number of extra layers to add to the middle stages
+    extra_layers = min(remainder, max(0, total_stages - 2))
+
+    splits = []
+    current_layer = 0
+    for i in range(total_stages - 1):
+        if i == 0 or i == total_stages - 2:
```

Comment:
`i == total_stages - 2` is the split point between 2nd last stage and 3rd last stage, which shouldn't get special treatment. I guess what you meant is 
```suggestion
        if i == 0:
```